components/nexus-core/src/main/java/org/sonatype/nexus/proxy/item/ChecksumReconciler.java

```diff
@@ -114,7 +129,46 @@ void reconcileItemChecksum(final Repository repo, final StorageFileItem item) th
         finally {
           uid.getLock().unlock();
         }
+
+        final String itemPath = item.getPath();
+        if (!itemPath.endsWith(".sha1") && !itemPath.endsWith(".md5")) {
```

Comment:
Since performance was mentioned as a concern, was just trying to keep us doing the least work possible - and I'm guessing that removing any need to do the initial 8-byte read would be worth a refactor. If it ran quickly on RSO/OSSRH, I'm not sure that we need to risk introducing problems by moving it.